presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractMapBlock.java

```diff
@@ -106,7 +108,8 @@ public Block copyPositions(List<Integer> positions)
         int[] hashTable = getHashTables();
         int[] newHashTable = new int[newOffsets[newOffsets.length - 1] * HASH_MULTIPLIER];
         int newHashIndex = 0;
-        for (int position : positions) {
+        for (int i = 0; i < length; ++i) {
```

Comment:
nit: You can iterate from `offset` to `offset + length` as done in other loops in this PR. There are several similar loops below if you want to update them all.

Reply:
In some places we use zero indexed `i` too, so it's not possible to change in all places, but I looked through again and changed indexing in some places where it was possible